brave/brave-core
# Add .torrent support

Fix https://github.com/brave/brave-browser/issues/855
## Submitter Checklist:


## Test Plan:
Manual Test:
1. Open a new tab and visit https://webtorrent.io/free-torrents, click torrent link of Sintel should leads to a torrent viewer page without start downloading
2. Click start download, torrent should start to be downloading
3. Click any link under save file should be able to download that specific file
4. Click on poster.jpg, should open a tab and show the image directly
5. Go back to the torrent downloading page and click on sintel.mp4, should open a tab to stream the media
(Note: It might take a while for the media to be ready to play.)
6. Close the media streaming tab and go back to the torrent downloading page
7. Go back to https://webtorrent.io/free-torrents tab and see if it also works by right click the torrent link to open it in a new tab

Automation Test:
`npm run test -- brave_browser_tests --filter=BraveContentBrowserClientTest.ReverseRewriteTorrentURL`
`npm run test -- brave_unit_tests --filter=BraveTorrentRedirectNetworkDelegateHelperTest.*`
`npm run test-unit`

## Reviewer Checklist:
